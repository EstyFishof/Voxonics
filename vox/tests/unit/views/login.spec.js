import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import User from '@/API/effects/User';
import Login from '../../../src/views/Login';

jest.mock('@/API/effects/User');

const localVue = createLocalVue();

localVue.use(Vuex);

describe('Login', () => {
    window.env = {
        BASE_URL: 'http://168.119.92.172/'
    };

    let getters;
    let store;
    let wrapper;

    beforeEach(() => {
        getters = {
            authStatus: () => ""
        };
        store = new Vuex.Store({ getters });
        wrapper = shallowMount(Login, { store, localVue, stubs: {Login: true} });
    });
    it('should  Vue instance', function () {
        expect(wrapper.vm).toBeTruthy();
    });
    it('Check snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('Input username', () => {
        expect(wrapper.find('.login-field').exists()).toBe(true);
    });
    it('Input password', () => {
        expect(wrapper.find('.password-field').exists()).toBe(true);
    });
    it('button login', () => {
        expect(wrapper.find('.btn-login').exists()).toBe(true);
    });
    it('login setData username', async () => {
        await wrapper.setData({ form: { internalNumber: 'admin' } });
        expect(wrapper.find('.login-field').element.value).toBe('admin');
    });
    it('login setData password', async () => {
        await wrapper.setData({ form: { password: 'admin' } });
        expect(wrapper.find('.password-field').element.value).toBe('admin');
    });
    it('submit form', async () => {
        const spy = jest.spyOn(wrapper.vm, 'submitForm');
        await User.login.mockResolvedValue('token');
        const result = await wrapper.vm.submitForm(wrapper.vm.form);
        expect(result).toEqual('token');
        expect(spy).toHaveBeenCalled();
        spy.mockRestore();
    });
});
