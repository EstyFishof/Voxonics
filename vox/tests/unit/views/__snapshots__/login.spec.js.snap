// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Check snapshot 1`] = `
<section
  class="container-form"
>
  <form
    class="form"
  >
    <div
      class="form-wrap"
    >
      <div
        class="logo"
      >
        <img
          alt="logo"
          src="../assets/img/logo.svg"
        />
      </div>
       
      <h4>
        Enter your credentials to access your account
      </h4>
       
      <div
        class="form-group"
      >
        <input
          class="login-field"
          placeholder="Internal Number"
          required="required"
        />
         
        <input
          class="password-field"
          placeholder="Enter password"
          required="required"
          type="password"
        />
         
        <div
          class="btn-container"
        >
          <button
            class="btn-login"
          >
            Sign in
          </button>
           
          <!---->
        </div>
      </div>
    </div>
  </form>
   
  <switchtheme-stub
    class="theme"
  />
</section>
`;
